File: python/7 kyu/379.py
# https://www.codewars.com/kata/54fb853b2c8785dd5e000957/train/python

# Passed

def chain(init_val, functions):
    for func in functions:
        init_val = func(init_val)
        
    return init_val

add10 = lambda x: x + 10
mul30 = lambda x: x * 30

output = chain(50, [mul30, add10])
print(output)

File: nodejs/7 kyu/379.js
// https://www.codewars.com/kata/5a651865fd56cb55760000e0/train/javascript

// Passed

function arrayLeaders(numbers) {
  let arrSum = numbers.reduce((prev, curr) => prev + curr, 0);
  let result = [];
  for (const num of numbers) {
    arrSum -= num;
    if (num > arrSum) {
      result.push(num);
    }
  }

  return result;
}

const output = arrayLeaders([16, 17, 4, 3, 5, 2]);
console.log(output);